### DISCUSSION TITLE
New to Zepp OS, some tips to start building my windsurfing app

### DISCUSSION THREAD

**Discussion by juacamgo** [zepp-health/zeppos-forum#295] [Q&A] [answered]
Hello, I am new to zepp OS developing. I have already developing skills (mostly Python and C#), and I'm willing to do a surfing/windsurfing app.

I've read the whole documentation and I'm starting to build my first app. Since all I want to do is check the weather, and get some things like wind speed, temperature, etc... Do you think the Fetch API is the best thing to start?

I've noticed the fetch sometimes fails in the example fetch api. Is that normal?

**Comment by silver-zepp** [comment, marked as answer]
Hey. Yeah, your approach with (probably) openweathermap is the most reasonable.
Speaking of Fetch API, this is also true. It's more robust on a physical device than on a simulator, but to be on the safer side you have to implement a retry mechanism to make sure that your requests succeed.

**Reply by juacamgo** [reply]
Hey, I've just implemented the first page, where I geolocate and use the data from openweathermap to get the data I need, I will mark your reply as answer.

Thank you!
